clippy_lints/src/doc/mod.rs

```diff
@@ -420,6 +421,37 @@ declare_clippy_lint! {
     "require every line of a paragraph to be indented and marked"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks if the first line in the documentation of items listed in module page is too long.
+    ///
+    /// ### Why is this bad?
+    /// Documentation will show the first paragraph of the doscstring in the summary page of a
+    /// module, so having a nice, short summary in the first paragraph is part of writing good docs.
+    ///
+    /// ### Example
+    /// ```no_run
+    /// /// A very short summary.
+    /// /// A much longer explanation that goes into a lot more detail about
```

Comment:
This seems like something that is an accident and catching that would definitely really nice. But I agree, that just limiting the first line to 80 characters is a bit too strict.

I haven't looked at the lint implementation. But could this lint use some heuristics like punctuation to limit the first line to either one single (longer) sentence or a few short sentences?

Reply:
It's really supposed to be one short sentence. I can move up the limit to 100. Also, good idea to check for punctuation. In this case we could suggest to add an empty doc line just after this one.

Reply:
I feel like there are cases where two sentences may explain it better as a summary. Like "for example" after it, where that might not make sense after a linebreak (without restating the above summary)

Reply:
I only look for `.`, `!` and `?`, so this case should never happen.